### ISSUE TITLE
Can't build atom-shell on W7-64

### ISSUE THREAD

**Issue by shrewdlogarithm** [atom/atom-shell#1187]
I'm getting an error which - although it's been posted here before - none of the posted solutions appear to work 

I'm guessing it's a VS-related issue but the 2012 error message makes zero sense when everything I've read says you only need VS 2013 and I only have that (I've NEVER had any other version of VS either)

I've run this in cmd and powershell and from the VS2013 command prompt (all from clean pulls) - I'm running node0.10.36 (32bit), python 2.7 and VS 2013 Comm Ed and everything seems to work (including node-gyp and I can "npm install runas" without errors at the command-line)

I also tried "npm config set msvs_version 2013" and "set GYP_MSVS_VERSION 2013" - same error

Output as follows

````
e:\python27\python.exe E:\webdev\atom-shell\atom-shell\script\update.py
E:\webdev\atom-shell\atom-shell\node_modules\atom-package-manager\bin\apm.cmd in
stall
Installing modules failed
> runas@2.0.0 install E:\webdev\atom-shell\atom-shell\spec\node_modules\runas
> node-gyp rebuild


E:\webdev\atom-shell\atom-shell\spec\node_modules\runas>node "E:\webdev\atom-she
ll\atom-shell\node_modules\atom-package-manager\node_modules\npm\bin\node-gyp-bi
n\\..\..\node_modules\node-gyp\bin\node-gyp.js" rebuild
Building the projects in this solution one at a time. To enable parallel build,
please add the "/m" switch.
C:\Program Files (x86)\MSBuild\Microsoft.Cpp\v4.0\V120\Microsoft.Cpp.Platform.ta
rgets(64,5): error MSB8020: The build tools for Visual Studio 2012 (Platform Too
lset = 'v110') cannot be found. To build using the v110 build tools, please inst
all Visual Studio 2012 build tools.  Alternatively, you may upgrade to the curre
nt Visual Studio tools by selecting the Project menu or right-click the solution
, and then selecting "Upgrade Solution...". [E:\webdev\atom-shell\atom-shell\spe
c\node_modules\runas\build\runas.vcxproj]

npm WARN package.json atom-shell-specs@0.1.0 No description
npm WARN package.json atom-shell-specs@0.1.0 No repository field.
npm WARN package.json atom-shell-specs@0.1.0 No README data
gypnpm ERR! Windows_NT 6.1.7601
npm ERR! argv "E:\\webdev\\atom-shell\\atom-shell\\node_modules\\atom-package-ma
nager\\bin\\\\node.exe" "E:\\webdev\\atom-shell\\atom-shell\\node_modules\\atom-
package-manager\\node_modules\\npm\\bin\\npm-cli.js" "--globalconfig" "E:\\webde
v\\atom-shell\\atom-shell\\node_modules\\atom-package-manager\\.apmrc" "--userco
nfig" "C:\\Users\\john\\.atom\\.apmrc" "install" "--target=0.21.0" "--arch=ia32"
 "--msvs_version=2012"
npm ERR! node v0.10.35
npm ERR! npm  v2.1.18
npm ERR! code ELIFECYCLE

npm ERR! runas@2.0.0 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the runas@2.0.0 install script 'node-gyp rebuild'.
npm ERR! This is most likely a problem with the runas package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-gyp rebuild
npm ERR! You can get their info via:
npm ERR!     npm owner ls runas
npm ERR! There is likely additional logging output above.
None
Traceback (most recent call last):
  File "script\bootstrap.py", line 127, in <module>
    sys.exit(main())
  File "script\bootstrap.py", line 35, in main
    update_atom_modules('spec')
  File "script\bootstrap.py", line 77, in update_atom_modules
    execute_stdout([apm, 'install'])
  File "E:\webdev\atom-shell\atom-shell\script\lib\util.py", line 152, in execut
e_stdout
    raise e
subprocess.CalledProcessError: Command '['E:\\webdev\\atom-shell\\atom-shell\\no
de_modules\\atom-package-manager\\bin\\apm.cmd', 'install']' returned non-zero e
xit status 1

**Comment by shrewdlogarithm** [comment]
One thing I would add - I mistakenly pasted the 'build' command instead of the bootstrap one and it seems to have built atom-shell without any errors - despite the bootstrap not completely as-above?

Certainly my app works with the newly build files (it had the "tray icon bug won't read from an asar" problem which isn't fixed in the current downloadable copy for Windows - so it's definately the newer build!?)

**Comment by shrewdlogarithm** [comment]
OK - traced this into the apm-cli.js file where it tries to determine what VS versions are available.

2 problems

1 - my VS isn't installed in the usual location (programfilesx86) because 'limited space on C'
2 - it seems NODE sometimes 'pads' env var names - see this grab from my debugging session

````
FP_NO_HOST_CHECK: 'NO',
'GYP_MSVS_VERSION ': ' 2013e',
HerokuPath: 'e:\\heroku',
````

Note the quotes/space after VERSION (2013e was me making sure the value changed - tho it's required by npm in some cases to indicate 'Community Edition' apparently?)

Bottom line is I just changed the getInstalledVisualStudioFlag function to return '2013' and it gets past that bit of the build - tho setting GYP_MSVS_VERSION without the spaces also works e.g.

set GYP_MSVS_VERSION=2013
not
set GYP_MSVS_VERSION = 2013

DOS allows the latter but node reads it as having a trailing space?

**Comment by shrewdlogarithm** [comment]
p.s. I notice there's a fair bit of unessential stuff left in the debug or release folders which I'm assuming isn't needed - a 'gen' and 'obj' folder, ninja files, manifest, lib, ilk, environment and 'gyp-win-tool'???
